test/ttmlir/Dialect/TTNN/simple_broadcast.mlir

```diff
@@ -1,12 +1,28 @@
-// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s | FileCheck %s
-module @jit_broadcast attributes {mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32} {
-  func.func public @main(%arg0: tensor<1xf32> {mhlo.layout_mode = "default"}, %arg1: tensor<512x512xf32> {mhlo.layout_mode = "default"}) -> (tensor<512x512xf32> {jax.result_info = "", mhlo.layout_mode = "default"}) {
-    // CHECK: %[[C:.*]] = "ttnn.empty"[[C:.*]]
-    // CHECK-NOT: %[[C:.*]] = "ttnn.broadcast"[[C:.*]]
-    %0 = tensor.empty() : tensor<512x512xf32>
-    %1 = "ttir.broadcast"(%arg0, %0) <{dimension = [1]}> : (tensor<1xf32>, tensor<512x512xf32>) -> tensor<512x512xf32>
-    %2 = tensor.empty() : tensor<512x512xf32>
-    %3 = "ttir.maximum"(%1, %arg1, %2) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<512x512xf32>, tensor<512x512xf32>, tensor<512x512xf32>) -> tensor<512x512xf32>
-    return %3 : tensor<512x512xf32>
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s| FileCheck %s
+module {
+  func.func @main(%arg0: tensor<1x16x1xf32>, %arg1: tensor<1x1x32xi32>) -> tensor<1x16x32xf32> {
+    // CHECK: %{{[0-9]+}} = "ttnn.repeat"(%{{[0-9]+}})
+    %0 = tensor.empty() : tensor<1x1x32xf32>
+    %1 = "ttir.typecast"(%arg1, %0) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<1x1x32xi32>, tensor<1x1x32xf32>) -> tensor<1x1x32xf32>
```

Comment:
why typecast

Reply:
This was copied from a front-end example, I have the test.